agent/agent.go

```diff
@@ -501,7 +501,7 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, initialTC *taskCon
 		Redacted:           tc.taskConfig.Redacted,
 		InternalRedactions: tc.taskConfig.InternalRedactions,
 	}
-	tc.taskConfig.TaskOutputDir = taskoutput.NewDirectory(tc.taskConfig.WorkDir, &tc.taskConfig.Task, redactorOpts, tc.logger)
+	tc.taskConfig.TaskOutputDir = taskoutput.NewDirectory(tc.taskConfig.WorkDir, &tc.taskConfig.Task, redactorOpts, tc.logger, a.tracer)
```

Comment:
we could avoid passing through the tracer by creating a new one in the package where it's used.
Something like [this](https://github.com/evergreen-ci/evergreen/blob/9376a7cf5e6e59506d8914413f89d554e31b130c/model/githubapp/otel.go#L12)

Reply:
I think we want to keep it so it is configured the same as the agent tracer so that is why I pass it down.

Reply:
Most of the configuration will remain the same because the tracer provider is in global state. The only thing that will change is the [library.name](https://ui.honeycomb.io/mongodb-4b/environments/production/datasets/evergreen/result/tqLDSCRy7n) attribute, which I think is what we want in order to see where in the code a trace is coming from.